### PR TITLE
[feature](cast) Support cast complex type to json

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
in this pr we do
1. support cast complex type to json which we can use array_agg/map_agg result into json_object
Issue Number: close #xxx

### CODE REVIEW

File: fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java [hunk 1]
```diff
@@ -834,6 +834,10 @@ public static boolean isImplicitlyCastable(Type t1, Type t2, boolean strict, boo
     }
 
     public static boolean canCastTo(Type sourceType, Type targetType) {
+        // In BE Code , we make any data type can cast to jsonb
+        if (targetType.isJsonbType()) {
+            return true;
```
Comment (morrySnow) [inline fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java:839]: agg_state, variant could cast to json? what we will get if cast ipv4 /ipv6 to json.

Comment (morrySnow) [inline fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java:839]: if support all type, u should add all type cast to json test case

**Comment by amorynan** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CastExpr.java [hunk 2]
```diff
@@ -115,6 +119,19 @@ public CastExpr(Type targetType, Expr e) {
...
+    public boolean checkMapKeyIsStringLikeForJson(Type complexType) {
+        if (complexType.isMapType()) {
+            return ((MapType) complexType).getKeyType().isStringType();
+        } else if (complexType.isArrayType()) {
+            return checkMapKeyIsStringLikeForJson(((ArrayType) complexType).getItemType());
+        } else if (complexType.isStructType()) {
+            for (StructField f : ((StructType) complexType).getFields()) {
+                return checkMapKeyIsStringLikeForJson(f.getType());
+            }
+        }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CastExpr.java:131]: should check in nereids, not here

File: fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java (hunk 1 above)
Comment (amorynan) [reply fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java:839]: add limits

**Comment by amorynan** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CastExpr.java (hunk 2 above)
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CastExpr.java]: done!

**Comment by amorynan** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**